src/helper/service-helper.ts

```diff
@@ -364,27 +366,21 @@ export class ZkWasmServiceHelper {
     return response;
   }
 
-  async addNewWasmImage(task: WithSignature<AddImageParams>) {
-    let response = await this.sendRequestWithSignature<AddImageParams>(
-      "POST",
-      TaskEndpoint.SETUP,
-      task,
-      true
-    );
+  async addNewWasmImage(task: WithSignature<RequiresNonce<AddImageParams>>) {
+    let response = await this.sendRequestWithSignature<
+      RequiresNonce<AddImageParams>
+    >("POST", TaskEndpoint.SETUP, task, true);
 
     if (this.endpoint.enable_logs) {
       console.log("get addNewWasmImage response:", response.toString());
     }
     return response;
   }
 
-  async addProvingTask(task: WithSignature<ProvingParams>) {
-    let response = await this.sendRequestWithSignature<ProvingParams>(
-      "POST",
-      TaskEndpoint.PROVE,
-      task,
-      true
-    );
+  async addProvingTask(task: SignatureRequest<RequiresNonce<ProvingParams>>) {
```

Comment:
Please review all POST type in this file and all of them may have a class to handle and we can remove all post APIs from this file. (Or some function can be in a class, for examle, addSubscription and addPayment can be in a class for payment related. setMaintenanceMode can be in a maintain class)

I see the reason maybe why you add SignatureRequest with WithOptionalNonce.
It is because:
```
async sendRequestWithSignature<T>(
    method: "GET" | "POST",
```

and only one use GET is 
```
async queryLogs(query: WithSignature<LogQuery>): Promise<string> {
    let logs = await this.sendRequestWithSignature(
      "GET",
      TaskEndpoint.LOGS,
      query
    );
```

In this area I find it is a little bit chaos as the sendRequestWithSignature() param changed to use SignatureRequest<T> type but I didn't see you change anything for the API, the input is still WithSignature<LogQuery>? which is not WithSignature<WithOptionalNonce<LogQuery>>?

So maybe WithOptionalNonce is not a good one?
We may do not need SignatureRequest?
For those POST functions (which should move to class and not expose), I can see we can just use WithSignature<RequiresNonce<T> type 

For this only queryLog which need signature, just use the WithSignature<T> as old? And sendRequestWithSignature can keep old inputs types?

We can discuss.